src/passes/publicFunctionSplitter/internalFunctionCallCollector.ts

```diff
@@ -0,0 +1,30 @@
+import {
+  FunctionCall,
+  FunctionCallKind,
+  FunctionDefinition,
+  Identifier,
+  MemberAccess,
+} from 'solc-typed-ast';
+import { AST } from '../../ast/ast';
+import { ASTMapper } from '../../ast/mapper';
+
+export class InternalFunctionCallCollector extends ASTMapper {
+  constructor(public internalFunctionCallSet: Set<FunctionDefinition>) {
+    super();
+  }
+
+  /*
+     This class collects all internal function calls in the contract
```

Comment:
This should explain why the class is needed. For example in this case something along the lines of how if there are not internal calls to a public function then it doesn't need to be split, which reduces the step count and bloat of the resulting cairo code. It should also go above the class rather than in the middle of it